src/overmap_ui.cpp

```diff
@@ -120,6 +121,317 @@ static const int npm_width = 3;
 /** Note preview map height without borders. Odd number. */
 static const int npm_height = 3;
 
+//forwards to cataimgui::draw_colored_text with automatic wrapping
+void overmap_sidebar::draw_sidebar_text( const std::string &original_text, const nc_color &color )
+{
+    cataimgui::draw_colored_text( original_text, color, float( str_width_to_pixels( width - 1 ) ) );
```

Comment:
It shouldn’t be necessary to call `str_width_to_pixels` here, or to keep track of an explicit width. ImGui knows how wide the window is already. You should be able to call `cataimgui::TextColoredParagraph` instead of `cataimgui_draw_colored_string`. It handles wrapping of everything except newline characters (`\n`) in the string without needing an explicit width to be passed in.